lib/automaticupgrades/channel.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2023 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package automaticupgrades
+
+import (
+	"context"
+	"net/url"
+
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/integrations/kube-agent-updater/pkg/maintenance"
+	"github.com/gravitational/teleport/integrations/kube-agent-updater/pkg/version"
+)
+
+// Channels is a map of Channel objects.
+type Channels map[string]*Channel
+
+func (c Channels) CheckAndSetDefaults() error {
+	var errs []error
+	var err error
+	for name, channel := range c {
+		err = trace.Wrap(channel.CheckAndSetDefaults(), "failed to create channel %s", name)
+		if err != nil {
+			errs = append(errs, err)
+		}
+	}
```

Comment:
Would you mind adding tests for this method?
It's not the first time I see a `CheckAndSetDefaults` not working as intended because it used a copy to the object and not the object itself.
Honestly, I'm not sure this case will fail but I would be more comfortable if we added some tests.